tests/system/source/main.d

```diff
@@ -64,15 +64,11 @@ int main (string[] args)
     const discovery_duration = 60.seconds;
     clients.enumerate.each!((idx, client) =>
     {
-        // TODO - fix node-0
-        if (idx > 0)
-        {
-            retryFor(client.getNodeInfo().ifThrown(NodeInfo.init)
-                .state == NetworkState.Complete,
-                discovery_duration,
-                format("%s %s has not completed discovery after %s.",
-                    PREFIX, nodeFromClientIndex(idx), discovery_duration * (idx + 1)));
-        }
+        retryFor(client.getNodeInfo().ifThrown(NodeInfo.init)
```

Comment:
it is a minor point, but can we not rely on the order of fields in the NetworkState and do something like this? 
retryFor(client.getNodeInfo().ifThrown((){NodeInfo ni; ni.state = NetworkState.Incomplete; return ni;}())

Reply:
Can you please explain why it is an improvement over the current implementation. I do not see the reason for the change you suggest.

Reply:
NodeInfo.init initializes NetworkState with NetWorkState.Incomplete, but just because Incomplete is defined as the first enum in NetworkState

Reply:
ok I see what you mean now. Thanks for the explanation.